Compilation error with SD and no WiFi library

Environment:

- Pico W
- Release 2.3.0
- Arduino IDE 1.8.19
- macOS Monterey (12.4)
- Apple M1 Pro

I have a simple SD test program which doesn't compile without including the WiFi library even though it's not used.
Including the WiFi library it compiles and runs without issues.

`
#include <SD.h>
#include <WiFi.h>


#define SPI_MOSI    7
#define SPI_MISO    4
#define SPI_SCK     6
#define SD_SELECT   5

void setup() {
  
  Serial.begin(9600);
  Serial.println("Initializing SD card...");

  delay(1000);
  
  SPI.setRX(SPI_MISO);    // MISO
  SPI.setCS(SD_SELECT);
  SPI.setSCK(SPI_SCK);
  SPI.setTX(SPI_MOSI);    // MOSI

  if (!SD.begin(SD_SELECT)) {
    Serial.println("Card failed, or not present");
  }
  else {
    Serial.println("Card initialized");
  }

  Serial.println("Ready");
}
`

Without including the WiFi library I get the following errors:

/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/lib/libpico.a(cyw43_ctrl.c.obj): in function `cyw43_ensure_up':
/home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ctrl.c:152: undefined reference to `cyw43_cb_tcpip_deinit'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ctrl.c:153: undefined reference to `cyw43_cb_tcpip_deinit'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/lib/libpico.a(cyw43_ctrl.c.obj): in function `cyw43_cb_process_async_event':
/home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ctrl.c:410: undefined reference to `cyw43_cb_tcpip_set_link_up'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ctrl.c:331: undefined reference to `cyw43_cb_tcpip_set_link_down'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ctrl.c:386: undefined reference to `cyw43_cb_tcpip_set_link_up'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ctrl.c:390: undefined reference to `cyw43_cb_tcpip_set_link_down'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/lib/libpico.a(cyw43_ll.c.obj): in function `cyw43_ll_process_packets':
/home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ll.c:1212: undefined reference to `cyw43_cb_process_ethernet'
/Users/fab/Library/Arduino15/packages/rp2040/tools/pqt-gcc/1.4.0-c-0196c06/bin/../lib/gcc/arm-none-eabi/10.3.0/../../../../arm-none-eabi/bin/ld: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/lib/libpico.a(cyw43_ll.c.obj): in function `cyw43_do_ioctl':
/home/earle/Arduino/hardware/pico/rp2040/pico-sdk/lib/cyw43-driver/src/cyw43_ll.c:1241: undefined reference to `cyw43_cb_process_ethernet'
collect2: error: ld returned 1 exit status
Multiple libraries were found for "SD.h"
 Used: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/libraries/SD
 Not used: /Applications/Arduino-1.8.19.app/Contents/Java/libraries/SD
Using library SD at version 2.0.0 in folder: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/libraries/SD 
Using library SDFS at version 0.1.0 in folder: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/libraries/SDFS 
Using library SPI at version 1.0 in folder: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/libraries/SPI 
Using library ESP8266SdFat at version 2.1.1 in folder: /Users/fab/Library/Arduino15/packages/rp2040/hardware/rp2040/2.3.0/libraries/ESP8266SdFat 
exit status 1
Error compiling for board Raspberry Pi Pico W.

--- a/cores/rp2040/sdkoverride/cyw43_arch_threadsafe_background.c
+++ b/cores/rp2040/sdkoverride/cyw43_arch_threadsafe_background.c
@@ -316,28 +316,34 @@ void cyw43_arch_poll() {
     //    }
 }
 
-
-static void no_lwip_fail() {
-    panic("You cannot use IP with pico_cyw43_arch_none");
-}
+#ifdef ARDUINO_RASPBERRY_PI_PICO_W
 void __attribute__((weak)) cyw43_cb_tcpip_init(cyw43_t *self, int itf);
 void cyw43_cb_tcpip_init(cyw43_t *self, int itf) {
+    (void) self;
+    (void) itf;
 }
 void __attribute__((weak)) cyw43_cb_tcpip_deinit(cyw43_t *self, int itf);
 void cyw43_cb_tcpip_deinit(cyw43_t *self, int itf) {
+    (void) self;
+    (void) itf;
 }
 void __attribute__((weak)) cyw43_cb_tcpip_set_link_up(cyw43_t *self, int itf);
 void cyw43_cb_tcpip_set_link_up(cyw43_t *self, int itf) {
-    no_lwip_fail();
+    (void) self;
+    (void) itf;
 }
 void __attribute__((weak)) cyw43_cb_tcpip_set_link_down(cyw43_t *self, int itf);
 void cyw43_cb_tcpip_set_link_down(cyw43_t *self, int itf) {
-    no_lwip_fail();
+    (void) self;
+    (void) itf;
 }
 void __attribute__((weak)) cyw43_cb_process_ethernet(void *cb_data, int itf, size_t len, const uint8_t *buf);
 void cyw43_cb_process_ethernet(void *cb_data, int itf, size_t len, const uint8_t *buf) {
-    no_lwip_fail();
+    (void) cb_data;
+    (void) itf;
+    (void) len;
+    (void) buf;
 }
-
+#endif
 
 #endif

--- a/cores/rp2040/sdkoverride/cyw43_arch_threadsafe_background.c
+++ b/cores/rp2040/sdkoverride/cyw43_arch_threadsafe_background.c
@@ -316,4 +316,28 @@ void cyw43_arch_poll() {
     //    }
 }
 
+
+static void no_lwip_fail() {
+    panic("You cannot use IP with pico_cyw43_arch_none");
+}
+void __attribute__((weak)) cyw43_cb_tcpip_init(cyw43_t *self, int itf);
+void cyw43_cb_tcpip_init(cyw43_t *self, int itf) {
+}
+void __attribute__((weak)) cyw43_cb_tcpip_deinit(cyw43_t *self, int itf);
+void cyw43_cb_tcpip_deinit(cyw43_t *self, int itf) {
+}
+void __attribute__((weak)) cyw43_cb_tcpip_set_link_up(cyw43_t *self, int itf);
+void cyw43_cb_tcpip_set_link_up(cyw43_t *self, int itf) {
+    no_lwip_fail();
+}
+void __attribute__((weak)) cyw43_cb_tcpip_set_link_down(cyw43_t *self, int itf);
+void cyw43_cb_tcpip_set_link_down(cyw43_t *self, int itf) {
+    no_lwip_fail();
+}
+void __attribute__((weak)) cyw43_cb_process_ethernet(void *cb_data, int itf, size_t len, const uint8_t *buf);
+void cyw43_cb_process_ethernet(void *cb_data, int itf, size_t len, const uint8_t *buf) {
+    no_lwip_fail();
+}
+
+
 #endif

